analyzers/src/SonarAnalyzer.Common/Helpers/SonarAnalysisContext.cs

```diff
@@ -94,12 +94,14 @@ internal SonarAnalysisContext(AnalysisContext context, IEnumerable<DiagnosticDes
         public bool ShouldAnalyzeGenerated(Compilation c, AnalyzerOptions options) =>
             ShouldAnalyzeGenerated(context.TryGetValue, c, options);
 
-        public static bool ShouldAnalyze(TryGetValueDelegate<bool> tryGetValue,
+        public static bool ShouldAnalyze(TryGetValueDelegate<bool> tryGetValueBool,
+                                         TryGetValueDelegate<ProjectConfigReader> tryGetValueProjectConfigReader,
```

Comment:
I would rename this to `tryGetProjectConfigReader`. `ProjectConfigReader` is the value.